src/main/java/org/terasology/metalrenegades/ai/actions/AddSocialComponentAction.java

```diff
@@ -35,10 +22,10 @@
 
     @Override
     public BehaviorState modify(Actor actor, BehaviorState result) {
-        SimpleSourceComponent socialSourceComponent = new SimpleSourceComponent(CitizenNeed.Type.SOCIAL);
+        SimpleSourceComponent socialSourceComponent = new SimpleSourceComponent();
+        socialSourceComponent.needType = CitizenNeed.Type.SOCIAL;
```

Comment:
is it really a bad smell to have additional constructors at all? I was about to introduce new ones for some components :rolling_eyes:  if you're using components programmatically and you need to create them with specific values, each user would end up writing a method to do that in one line ...

For instance, compare https://github.com/Terasology/Xmas/blob/e823584a898480cbd7889b76deafe728144696ff/src/main/java/org/terasology/xmas/player/SurpriseStartingInventory.java#L80-L93

Reply:
constructors define an initial state which is better then just letting people submit anything.  

for redux you have an state --> action --> mutations.